packages/toolpad-app/src/server/data.ts

```diff
@@ -529,14 +529,19 @@ export async function loadRenderTree(appId: string, version: VersionOrPreview =
   return appDom.createRenderTree(await loadDom(appId, version));
 }
 
-export async function duplicateApp(id: string, name: string): Promise<AppMeta> {
+export async function duplicateApp(
+  id: string,
+  name: string,
+  suppliedName?: boolean,
```

Comment:
I don't understand this parameter.

Reply:
This is essentially to allow reusing the function in two scenarios:

1. When the user doesn't supply a name when duplicating (duplicating in the list view)
2. When we have a user supplied name while duplicating the app (duplicating in the editor view)

Reply:
On second thought, we can unify both these experiences and then get rid of the need of having to do this.